### PR TITLE
Documentation Updates!

### PR DESCRIPTION
**Pull request by sheryjoe**
This PR includes the resolution of #649 #650, #648 and MORE and MORE, see below.  

What is left is 
- the documentation for the cutting plane use case @HeavenlyBerserker please provide needed information as mentioned in my email.
- feature map documentation in "what is new", #712
- @akenmorris please watch out for any edits needed in light of new Studio updates in #712 

This PR includes the following:
- Consolidated mkdocs.md and autodoc.md -> docs.md.  
- Rewrote analyze.md and optimize.md with more discussions and illustrations
- Added latex support to our docs for equations so we can use latex and render it very nicely (as opposed to low-res images of equations).
- Added support for call-outs and added instructions to use it in docs.md
- Additions to the home page
    - Added announcement to the get-togethers
    - Why ShapeWorks?
    - ShapeWorks vs other tools on a simple example
- Added "What is new" section with illustrations for 
    - dynamic loading (in Studio)
    - surface reconstruction (in Studio)
    - particles on meshes
    - consolidated commands
    - memory optimizations
    - Live particles movement (in Studio)
    - Quantitative evaluation for shape models 
- Added selected success stories to demonstrate ShapeWorks impact
- Some theme customization, e.g., colored font for sections for more readability
- Fixed DocumentationUtils and added runnable script to avoid having to auto-generate documentation within a python shell
- Added team picture

**Steps:**

- checkout mkdocs branch
- run mkdocs serve from the root directory (containing mkdocs.yml)
- open and review the following:    
    - http://127.0.0.1:8000/dev/
    - http://127.0.0.1:8000/getting-started/sw-stories/
    - http://127.0.0.1:8000/new/new-studio/
    - http://127.0.0.1:8000/new/openvdb/
    - http://127.0.0.1:8000/new/sw-meshes/
    - http://127.0.0.1:8000/new/shapeworks-api/
    - http://127.0.0.1:8000/new/ssm-eval/
    - http://127.0.0.1:8000/dev/docs/
    - http://127.0.0.1:8000/dev/workflow/analyze/
    - http://127.0.0.1:8000/dev/workflow/optimize/
    - http://127.0.0.1:8000/about/team/

**Note:** mkdocs branch will be used for editing/adding docs and I added a branch rule to prevent its deletion.

### CODE REVIEW

**Comment by sheryjoe** [comment]
@akenmorris please review and test the packages scripts as I changed the autodoc.

File: docs/dev/contribute.md [hunk 1]
```diff
@@ -6,7 +6,8 @@
 ShapeWorks uses `git` for managing source code so changes can be integrated from multiple people. Here are some of the basics to check out the repository, make and share modifications, and keep track of all your changes along the way.
 The ShapeWorks repositories are stored on [GitHub](https://github.com/SCIInstitute/ShapeWorks). 
 
-:fa-info-circle: _**Important:**_ If you want to contribute, first _make sure you have a developer account_.  
+!!! imporant
+    If you want to contribute, first _make sure you have a developer account_.
```
Comment (riddhishb) [inline docs/dev/contribute.md:10]: spell check

File: docs/new/ssm-eval.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
+!!! note "Generalization"
+    The model should be able to generalize from the examples given in the training set, hence describing any valid instance of the class of object, not just those seen in the training set.
+    
+![Generalization metric](../img/new/eval_gen.png)
+
+!!! note "Specificity"
+    This is the requirement that the model can only represent valid instances of the class(es) of objects presented in the training set. Hence, the model is specific for this training set.
+
```
Comment (riddhishb) [inline docs/new/ssm-eval.md:25]: Should we add something about what is a good balance between compactness, generalization and specificity as there is a tradeoff between these numbers, the example that compact model =0.99 is misleading a little.

File: docs/workflow/optimize.md [hunk 3]
```diff
@@ -1,104 +1,192 @@
...
-
-Correspondence positions are optimized by gradient descent on an energy function that balances the negative entropy of 
-the distribution of particles in configuration space with the positive entropy of the distribution of the configurations
-in shape space. The method is to consider ![equation](../img/workflow/z_k.png)
- both as observations on a dM × 1 vector random
-variable Z and as N samples of M observations on N , d × 1 vector random variables ![equation](../img/workflow/x_k.png)
- . The optimization to establish correspondence minimizes the energy function
+## Particle-based Shape Modeling
+
+ShapeWorks constructs *statistically optimal anatomical mapping across different shape samples* by automatically computing a dense set of corresponding landmark positions on a set of anatomy segmentations or surface meshes and *does not rely on any specific surface parameterization*.
```
Comment (riddhishb) [inline docs/workflow/optimize.md:5]: add geometrically consistent in this sentence basically define what is "in correspondence"

**Review by riddhishb — COMMENTED** [review]
Very minor comments in overall clarification but looks good to go otherwise to me

**Review by medakk — APPROVED** [review]
Looks great! I don't know why but the links in the PR (like http://127.0.0.1:8000/dev/) did not work for me. I used to sidebar to get to the pages

File: docs/new/ssm-eval.md (hunk 2 above)
Comment (sheryjoe) [reply docs/new/ssm-eval.md:25]: Good point. Added

File: docs/workflow/optimize.md (hunk 3 above)
Comment (sheryjoe) [reply docs/workflow/optimize.md:5]: done

File: docs/dev/build.md [hunk 4]
```diff
@@ -86,7 +90,8 @@ Download and install [[Visual Studio 2019]](https://visualstudio.microsoft.com/)
 #### Anaconda
 Download and install [[Anaconda]](https://www.anaconda.com/).
```
[the remark this answers is not in the document]
Comment (sheryjoe) [reply docs/dev/build.md:91]: @akenmorris @cchriste will miniconda be sufficient for our needs here?

**Comment by sheryjoe** [comment]
@jadie1 @medakk @akenmorris @cchriste @archanasri @riddhishb @HeavenlyBerserker 

I have addressed all comments, restructured release notes, and added documentation for the cutting planes use cases. PLEASE review and let me know your comments and edits. 

This should be merged VERY SOON.

File: docs/about/release-notes.md [hunk 5]
```diff
@@ -1,5 +1,69 @@
...
+
+
+#### ShapeWorks Back-end
+
+* **ShapeWorks directly on meshes:** ShapeWorks now supports particle optimization directly on triangular surface meshes. This mesh support also enables working with open meshes without additional user inputs. See [ShapeWorks Directly on Meshes](http://sciinstitute.github.io/ShapeWorks/new/sw-meshes) for more details. For exemplar use cases, see [Femur Mesh: SSM directly from meshes](http://sciinstitute.github.io/ShapeWorks/use-cases/femur-mesh) and [Lumps: SSM directly from meshes](http://sciinstitute.github.io/ShapeWorks/use-cases/lumps) to learn how to get started. *This mesh support allows for significant memory savings (9.2GB to 53MB in the femur use case)*.
+
+* **New ShapeWorks API:** Consolidation of image-based and segmentation-based grooming tools that creates a stable and reusable API making it much easier and more flexible for users to groom their datasets. This includes a full complement of unit tests. See [ShapeWorks API](http://sciinstitute.github.io/ShapeWork/new/shapeworks-api) for more details.
```
Comment (medakk) [inline docs/about/release-notes.md:19]: typo in url

File: docs/about/release-notes.md [hunk 6]
```diff
@@ -1,5 +1,69 @@
...
+
+* **Enable only-shape data for the femur use case:** Femur use case demonstrates the processing workflow starting from surface meshes and can now be run without grooming images (in case they are not available). To groom with images, use the `--start_with_image_and_segmentation_data` tag. If this tag is not included, images will not be used in grooming.
+
+#### ShapeWorks Back-end
+
+* **Improved constrained particle optimization:** An improved implementation for cutting planes that support single and multiple cutting planes per sample has been added. This can be used in modeling scenarios where statistical modeling/analysis is needed for a region-of-interest on the anatomy/object-class at hand without having to affect the input data. See [Ellipsoid: Cutting Planes](http://sciinstitute.github.io/ShapeWorks/use-cases/ellipsoid-cutting-planes) and [](http://sciinstitute.github.io/ShapeWorks/use-cases/femur-cutting-planes) for exemplar use cases.
```
Comment (medakk) [inline docs/about/release-notes.md:48]: missing text for femur cutting planes link

File: docs/about/release-notes.md (hunk 5 above)
Comment (sheryjoe) [reply docs/about/release-notes.md:19]: Fixed. Thanks!

File: docs/about/release-notes.md (hunk 6 above)
Comment (sheryjoe) [reply docs/about/release-notes.md:48]: Fixed. Thanks!

**Review by HeavenlyBerserker — APPROVED** [review]
Yup

**Comment by sheryjoe** [comment]
@jadie1 @medakk @akenmorris @cchriste @archanasri @riddhishb @HeavenlyBerserker

Please let me know your comments on this graphical abstract for 5.5. This is a screen capture so ignore the red line under shapeworks.

image

**Comment by sheryjoe** [comment]
Another version

image

**Comment by medakk** [comment]
Its a little cluttered IMO, can't tell very well what images are related to each other. Maybe make it list/grid with clear demarcations(or lots of space in between?)

**Comment by sheryjoe** [comment]
@medakk good point. how about this one?

image

**Comment by sheryjoe** [comment]
> Looks great!

@jadie1 do you approve the PR?

**Comment by archanasri** [comment]
@sheryjoe the latest graphic looks better...less cluttered :)

**Comment by medakk** [comment]
@sheryjoe yup this looks good! maybe put the "ShapeWorks" under/besides the logo instead of on top of it

**Comment by sheryjoe** [comment]
> @sheryjoe yup this looks good! maybe put the "ShapeWorks" under/besides the logo instead of on top of it

Doing that will break the symmetry of the grid :(

**Comment by akenmorris** [comment]
Nice!

**Comment by HeavenlyBerserker** [comment]
👍

**Comment by sheryjoe** [comment]
Happy merging! Thanks all for your comments and feedback!

**Pull request merged**